### PR TITLE
Rewrote README.MD for clarity & comprehension

### PR DESCRIPTION
**Pull request by ariccio**
```
Squashed commit of the following:
commit 9ccd5fb0b5937915c0d20efaeecbadfa1fca7b10
Author: Alexander Riccio <alexander@riccio.com>
Date:   Mon Jun 15 20:05:17 2015 -0400

    Tweaked README.md to be consistent for literals.

    Specifically literal filenames.

commit cb901d331f225b920338e9dfa67bff706a8dd27b
Author: Alexander Riccio <alexander@riccio.com>
Date:   Mon Jun 15 20:03:00 2015 -0400

    fixed up README.md header formatting

commit 19f7ed9f1687787acfc46ecc0a8e76ddeab18a21
Author: Alexander Riccio <alexander@riccio.com>
Date:   Mon Jun 15 19:44:57 2015 -0400

    Rewrote README.MD for clarity & comprehension

Thank you: http://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git/5190323#5190323
```

I rewrote part of README.MD so that it'd be easier for newcomers to get started. It's more detailed, and (I think) better formatted.

Screenshot at 67% zoom:
notepad betterreadme
*imagine that these two screenshots are joined perfectly*
notepad betterreadme_2

### CODE REVIEW

**Comment by Croydon** [comment]
:+1:

**Comment by ariccio** [comment]
Sooo.... no activity?

**Comment by ariccio** [comment]
@milipili Should I open up an issue for this, now that the new policy requires it?

File: README.md [hunk 1]
```diff
@@ -1,38 +1,53 @@
-What is Notepad++ ?
-===================
+# What is Notepad++ ?
```
Comment (milipili) [inline README.md]: Please keep the old style.

File: README.md [hunk 2]
```diff
@@ -1,38 +1,53 @@
...
 
 [![Join the chat at https://gitter.im/notepad-plus-plus/notepad-plus-plus](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/notepad-plus-plus/notepad-plus-plus?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 
 Notepad++ is a free (free as in both "free speech" and "free beer") source code editor and Notepad replacement that supports several programming languages and natural languages. Running in the MS Windows environment, its use is governed by GPL License.
 
 
-To build Notepad++ package from source code:
-============================================
+## To build Notepad++ from source:
```
Comment (milipili) [inline README.md]: Idem here

File: README.md [hunk 3]
```diff
@@ -1,38 +1,53 @@
...
+## To build `notepad++.exe`:
+
+Open `PowerEditor\visual.net\notepadPlus.vcproj`, and build Notepad++ [like a normal Visual Studio project](https://msdn.microsoft.com/en-us/library/7s88b19e.aspx).
+
+## To build `SciLexer.dll` with boost:
 
-SciLexer.dll:
-From version 6.0, SciLexer.dll comes with release contains boost's PCRE (Perl Compatible Regular Expressions) feature.
-Therefore [Boost](http://www.boost.org/) is needed to compile Scintilla in order to have PCRE support.
+From version 6.0, SciLexer.dll comes with release contains boost's PCRE (Perl Compatible Regular Expressions) feature. [`C++11`'s regex](http://en.cppreference.com/w/cpp/regex) is **not** PCRE, thus [Boost](http://www.boost.org/) is needed to compile Scintilla.
```
Comment (milipili) [inline README.md]: We don't care of `std::regex` here. Maybe in a developer's FAQ

File: README.md [hunk 4]
```diff
@@ -1,38 +1,53 @@
...
+    - From the IDE, you can do this by right clicking on a file in Solution Explorer, and clicking "Open Command Prompt". This will open up a command prompt with all the proper environment variables.
+    - From the Windows Start screen/menu, type `Developer Command Prompt for VS2013`, and click/select the result.
+    - From an *already open* command prompt, run `vcvarsall.bat` (e.g. "C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\vcvarsall.bat").
+ 2. Change directory (`cd` or `pushd`) to `scintilla\win32\`
+ 3. Build `SciLexer.dll` with one of the following commands:
+    - `nmake NOBOOST=1 -f scintilla.mak`         (normal build)
+    - `nmake NOBOOST=1 DEBUG=1 -f scintilla.mak` (debugging build)
+ 4. Copy `SciLexer.dll` from `scintilla\bin\` to the same directory as `notepad++.exe`.
+    - For the `Unicode Release` configuration, the output directory (where `notepad++.exe` is) is `PowerEditor\bin\`.
+    - For the `Unicode Release` configuration, the output directory (where `notepad++.exe` is) is `PowerEditor\visual.net\Unicode Debug\`.
```
Comment (milipili) [inline README.md]: You may want to say `Unicode Debug` configuration ?

File: README.md [hunk 5]
```diff
@@ -1,38 +1,53 @@
...
+You can build Notepad++ with *or* without boost - The release build of Notepad++ is built **with** boost.
 
 There should be several ways to generate Notepad++ binaries, here we show you only the way with which Notepad++ official releases are generated.
-* notepad++.exe: Visual Studio 2013
-* SciLexer.dll: Visual Studio 2013 (with nmake)
 
-notepad++.exe:
-Double click on `PowerEditor\visual.net\notepadPlus.vcproj` to launch Notepad++ project in Visual Studio, then build it with the mode you want, that's it.
+There are two components that need to be built separately:
+- `notepad++.exe`: (depends on `SciLexer.dll`)
```
Comment (milipili) [inline README.md]: Maybe we could add here depends on `ScopLexer.dll` form the Scintilla project.

**Comment by milipili** [comment]
It falls for me under the "user documentation" section, so it's fine. However please see my remarks and provide only one commit for this.

**Comment by ariccio** [comment]
Fair enough. I don't agree with the header styling, but it's not important, and I'll change it back to the old style.

I'll jump through some git hoops, make the requested changes, and open up a separate commit.

File: README.md (hunk 4 above)
Comment (ariccio) [inline README.md]: Oops. Yes.

File: README.md (hunk 2 above)
Comment (ariccio) [inline README.md]: Do you also want the old style for "To build `notepad++.exe`:", "To build `SciLexer.dll` with boost:", and "To build `SciLexer.dll` *without* boost:"?

Comment (milipili) [inline README.md]: I don't have any preferences for that. It was just for `h1` and `h2` tags. And ~80 columns in the same time if possible.

File: README.md (hunk 5 above)
Comment (ariccio) [inline README.md]: `ScopLexer.dll`? What is that? I've never heard of that...

**Pull request closed without merging**